### ISSUE TITLE
Highlighting: Don't highlight inside highlight

### ISSUE THREAD

**Issue by elyahw** [gsantner/markor#2417]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**. Not a [question or feature request](https://github.com/gsantner/markor/issues/new/choose).
- [X] The topic is **not** already reported at [Issues](https://github.com/gsantner/markor/issues?q=#js-issues-search). _(I've searched it)_.
- [X] Markor **is** up to date. See [Releases](https://github.com/gsantner/markor/tags) for the latest version. Updates are available from [F-Droid](https://f-droid.org/en/packages/net.gsantner.markor/) and GitHub.
- [X] The bug is still present in the latest development version (git master). (Please [download](https://nightly.link/gsantner/markor/workflows/build-android-project/master) and try the test version of Markor, named **Marder**. Don't worry; Markor and Marder appear as completely separate applications. You can install both side-by-side, and Markor settings are not touched. In case the issue is resolved there, you don't need to create a bug report. The change will be part of the next Markor update.)

### Description
Simply put: The url in this image should be orange, not blue, as it is within another highlight
![Screenshot from 2024-09-17 11:16:08](https://github.com/user-attachments/assets/77698156-4181-49dd-95d2-8ad482b6a6e9)

### Steps to reproduce
 Just type: `# this is a title https://www.github.com/gsantner/markor`

The url in the line above should be orange, not blue.

### Information
Android version: 13
Device: Android Pixel A4
App Version: v2.12.5

### Source
GitHub Releases

**Comment by harshad1** [comment]
I disagree. The url part should be blue as it is a url. We want things which can be navigated to (potentially) to stand out.
